valhalla/midgard/encoded.h

```diff
@@ -1,22 +1,110 @@
-#ifndef VALHALLA_MIDGARD_ENCODED_H_
-#define VALHALLA_MIDGARD_ENCODED_H_
-
-#include <valhalla/midgard/shape_decoder.h>
+#pragma once
 
 #include <cmath>
+#include <stdexcept>
 #include <string>
 #include <type_traits>
 #include <vector>
 
+// we store 6 digits of precision in the tiles, changing to 7 digits is a breaking change
+// if you want to try out 7 digits of precision you can uncomment this definition
+//#define USE_7DIGITS_DEFAULT
```

Comment:
this breaks compatibility but, for testing, if you want you can try out 7 digits of precision with this simple preprocessor macro